modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/QueryMemoryManager.java

```diff
@@ -80,6 +77,17 @@
     /** Memory reserved by running queries. */
     private final AtomicLong reserved = new AtomicLong();
 
+    /**
+     * Defines an action that occurs when the memory limit is exceeded. Possible variants:
+     * <ul>
+     * <li>{@code true} - exception will be thrown.</li>
+     * <li>{@code false} - intermediate query results will be spilled to the disk.</li>
+     * </ul>
+     *
+     * Default: false.
+     */
+    private final boolean failOnMemLimitExceed;
```

Comment:
This duplicates similat field in MemoryTracker.
Let's left only one of them as they can't be changed independently and most likely can't be changed at all.

Reply:
It should be initialized each time on creation. See explanation above.